bndtools.m2e/src/bndtools/m2e/Activator.java

```diff
@@ -1,32 +1,59 @@
 package bndtools.m2e;
 
-import org.eclipse.m2e.core.internal.MavenPluginActivator;
-import org.eclipse.m2e.core.internal.project.registry.MavenProjectManager;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.embedder.IMaven;
+import org.eclipse.m2e.core.project.IMavenProjectRegistry;
 import org.osgi.annotation.bundle.Header;
+import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
 
 import aQute.bnd.osgi.Constants;
 
 @Header(name = Constants.BUNDLE_ACTIVATOR, value = "${@class}")
 public class Activator implements BundleActivator {
 
+	/**
+	 * M2E version 2 has implicit support for whiteboard listeners and registers
+	 * services. This activator helps us to work in both situations.
+	 */
+	private boolean								isM2E_v2;
+
 	private MavenProjectChangedListenersTracker projectChangedListenersTracker;
 
 	@Override
 	public void start(BundleContext context) throws Exception {
-		projectChangedListenersTracker = new MavenProjectChangedListenersTracker();
-		MavenProjectManager projectManager = MavenPluginActivator.getDefault()
-			.getMavenProjectManager();
-		projectManager.addMavenProjectChangedListener(projectChangedListenersTracker);
+		Bundle bundle = FrameworkUtil.getBundle(MavenPlugin.class);
+		isM2E_v2 = bundle.getVersion()
+			.getMajor() == 2;
```

Comment:
Should this be 
```suggestion
			.getMajor() >= 2;
```

for some level of future proofing>

Reply:
I did wonder, but the boolean would need to be renamed too. 
The whole codebase needs a refactor, and fundamentally this class will be removed completely once we have 2.x as a base. I therefore leaned towards “make it work for now”.

Reply:
It’s also worth noting that the ceiling for the import is capped at `3.0.0)` in the bnd file, so `> 2` can’t occur without changing that.